src/finalisers/esm.ts

```diff
@@ -83,7 +83,13 @@ export default function esm(
 	const exportBlock: string[] = [];
 	const exportDeclaration: string[] = [];
 	exports.forEach(specifier => {
-		if (specifier.exported === 'default') {
+		if (specifier.shim) {
+			if (specifier.exported === 'default') {
+				exportBlock.push(`export ${specifier.exported} null;`);
+			} else {
+				exportBlock.push(`export const ${specifier.exported} = null;`);
+			}
+		} else if (specifier.exported === 'default') {
```

Comment:
I wonder if we could shift these types of logic out of the finalisers by rendering a line like:

```js
export const $$shim = null;
```

as part of the core rendering process, then setting the `specifier.local` value to `$$shim`.

Reply:
I'm not sure I follow. Wouldn't it get really complicated if there were multiple missing exports for a given modules? Like:

```js
import { missing1, missing2 } from './foo';

missing1('bar');
missing2('baz');
```

Reply:
Sorry I should have clarified rendering just a `${varOrConst} $$shim = null;`, so that in `foo` itself the output would be:

```js
const $$shim = null;

export { $$shim as missing1, $$shim as missing 2 }
```

Let me know if that makes sense?

Reply:
It does now. But aren't you worried about adding a var after the deconflicting step? Any name we choose, like $$shim, would possibly conflict with an existing $$shim var.

Reply:
I made the change, but the conflict issue still worries me.

Reply:
`$$shim` can be added to the list of names to deshadow in the `setIdentifierResolutions` function of `Chunk` to ensure proper handling of this.

Reply:
I'm now running into the problem where my output is

```js
const $$shim$1 = null;

var $$shim = null;
export { $$shim$1 as $$shim, $$shim };
```

This is a dependency and an entry point. My existing unused `$$shim` was renamed to `$$shim$1` because of the deshadowing, but is still exported as `$$shim` because of entry point. Now this conflicts with my shimmed export `$$shim`. It's almost like we need to reverse the renaming by leaving existing `$$shim`s as is, and instead do something like `chunk.generateUniqueVarAndExportName()` for the shimmed export name. Am I on the right track?

Reply:
I got something working.